lib/Doctrine/DBAL/Driver/Result.php

```diff
@@ -65,4 +62,31 @@ public function fetchAllAssociative(): array;
      * @throws DriverException
      */
     public function fetchFirstColumn(): array;
+
+    /**
+     * Returns the number of rows affected by the last DELETE, INSERT, or UPDATE statement
+     * executed by the corresponding object.
+     *
+     * If the last SQL statement executed by the associated Statement object was a SELECT statement,
+     * some databases may return the number of rows returned by that statement. However,
+     * this behaviour is not guaranteed for all databases and should not be
+     * relied on for portable applications.
+     *
+     * @return int The number of rows.
+     */
+    public function rowCount();
+
+    /**
+     * Returns the number of columns in the result set
+     *
+     * @return int The number of columns in the result set represented
+     *                 by the PDOStatement object. If there is no result set,
+     *                 this method should return 0.
```

Comment:
```suggestion
     *             by the PDOStatement object. If there is no result set,
     *             this method should return 0.
```

Reply:
One day we will also scrape out all those PDO references from non-PDO related code.